### DISCUSSION TITLE
What happens if connection limit is reached and pool_timeout is set to 0?

### DISCUSSION THREAD

**Discussion by cesarvspr** [prisma/prisma#25309] [Q&A]
### Question

Trying to gather info on this.

I think all new connections will be dropped. but not sure.

**Comment by ludralph** [comment]
Hi @cesarvspr 👋 

When all available connections in the connection pool are in use, new queries cannot immediately acquire a connection. When the `pool_timeout` is set to 0, this means that the Prisma Query Engine will not enforce a timeout for acquiring a connection from the pool. Instead of throwing an error after a waiting period, queries are placed in a queue and remain there until a connection becomes available. Essentially, the timeout is disabled, and Prisma will wait indefinitely until a connection is available to process the query. See this [section](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/databases-connections#disabling-the-pool-timeout) of the documentation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
